catroid/src/main/java/org/catrobat/catroid/formulaeditor/FormulaElement.java

```diff
@@ -914,7 +916,9 @@ private Double interpretOperatorEqual(Object left, Object right) {
 		}
 		if (left instanceof String && right instanceof Double) {
 			try {
-				int compareResult = Double.valueOf((String) left).compareTo((Double) right);
+				Double tempLeft = Double.parseDouble((String) left);
+				Double tempRight = (Double) right;
+				int compareResult = getCompareResult(tempLeft, tempRight);
```

Comment:
You're only handling 2 of the 4 cases where numbers are compared in this method.
Once handling all cases, try to improve this method as refactoring step (e.g. remove duplicate code).